js/components/dataset/resource/form.vue

```diff
@@ -243,7 +244,23 @@ export default {
             }]
         },
         fields() {
-            return this.generic_fields.concat(this.file_fields);
+            return this.generic_fields.concat(this.schema_field).concat(this.file_fields);
```

Comment:
The schema field will not be added to the form if the schema catalog is empty / is not set.